contracts/upgradeable_contracts/arbitrary_message/MessageDelivery.sol

```diff
@@ -2,22 +2,48 @@ pragma solidity 0.4.24;
 
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 import "./BasicAMB.sol";
+import "./VersionableAMB.sol";
+import "../../libraries/Bytes.sol";
 
-contract MessageDelivery is BasicAMB {
+contract MessageDelivery is BasicAMB, VersionableAMB {
     using SafeMath for uint256;
 
-    function requireToPassMessage(address _contract, bytes _data, uint256 _gas) public {
+    /**
+    * @dev Requests message relay to the opposite network
+    * @param _contract executor address on the other side
+    * @param _data calldata passed to the executor on the other side
+    * @param _gas gas limit used on the other network for executing a message
+    */
+    function requireToPassMessage(address _contract, bytes _data, uint256 _gas) public returns (bytes32) {
         require(_gas >= getMinimumGasUsage(_data) && _gas <= maxGasPerTx());
-        emitEventOnMessageRequest(abi.encodePacked(msg.sender, _contract, _gas, uint8(0x00), _data));
+
+        uint256 chainId = _chainId();
+        bytes20 bridgeId = bytes20(keccak256(abi.encodePacked(chainId, address(this))));
```

Comment:
```suggestion
        bytes32 bridgeId = keccak256(abi.encodePacked(chainId, address(this))) & 0x00000000ffffffffffffffffffffffffffffffffffffffff0000000000000000;
        uint64 nonce = _nonce();
        _setNonce(nonce + 1);
        
        bytes32 messageId = bytes32(ENCODED_BRIDGE_VERSION) << 224 | bridgeId | bytes32(nonce);
```

Reply:
Seems that it makes equivalent things but consumes less gas

Reply:
Nice improvement. Added in a805643. I have also changed the constant type to bytes32, so that we don't need to perform bit shift every time.